crates/parcel_filesystem/src/lib.rs

```diff
@@ -53,3 +57,63 @@ pub trait FileSystem {
   fn is_file(&self, path: &Path) -> bool;
   fn is_dir(&self, path: &Path) -> bool;
 }
+
+impl<'a, Fs: FileSystem> FileSystem for &'a Fs {
```

Comment:
Why is this needed? I don't really understand it

Reply:
This is here to support the following cases - https://github.com/parcel-bundler/parcel/pull/9827/files/58b162d26553a1f9f39e6434ea96dce4fe351135#diff-4ccc502f9a5e2b717a497747f4dd20b4a48b43e51f6f5476a9db347f9756bbbcR177

```
    let child = MockFileSystem::new();
    let _tracking = TrackingFileSystem::new(&child);
    let child: FileSystemRef = Arc::new(MockFileSystem::new());
    let _tracking = TrackingFileSystem::new(&child);
    let _tracking = TrackingFileSystem::new(child);
```

We'd want to support `TrackingFileSystem::new(&other_fs)` because the way things will look like could be by borrowing fs.

This allows `Arc<dyn FileSystem>` to be `Arc<MockFileSystem>` but also `Arc<&MockFileSystem>` or for `TrackingFileSystem<Fs: FileSystem>` to be `TrackingFileSystem<Arc<dyn FileSystem>>` but also `TrackingFileSystem<MockFileSystem>` or `TrackingFileSystem<&MockFileSystem>`.

I can remove these blanket implementations from this diff

Reply:
Happy to keep it, makes sense now